cql3/prepare_context.cc

```diff
@@ -27,14 +27,19 @@ std::vector<lw_shared_ptr<column_specification>> prepare_context::get_variable_s
     return std::move(_variable_specs);
 }
 
-std::vector<uint16_t> prepare_context::get_partition_key_bind_indexes(const schema& schema) const {
+std::vector<uint16_t> prepare_context::get_partition_key_bind_indexes() const {
+    if (!_schema) {
+        return {};
```

Comment:
I found this if() confusing and I'm still not sure I fully undersood it. Is the case where there is no schema the case where we have a batch of multiple tables? Do we deliberately want to return an empty vector in that case? If so, maybe it should be explained why in some comment.
Also, it's not obvious to me that the `!_schema` only happens when we have multiple tables - is it not possible that because of a bug in the code, we forgot to call set_schema() at all?

Reply:
One case is BATCH with zero statements, another is a statement that does not refer to a schema.

I'll think of a safer way to do this.